src/Storages/StorageLog.cpp

```diff
@@ -214,8 +214,7 @@ class LogSink final : public SinkToStorage
         , storage(storage_)
         , metadata_snapshot(metadata_snapshot_)
         , lock(std::move(lock_))
-        , marks_stream(
-            storage.disk->writeFile(storage.marks_file_path, 4096, WriteMode::Rewrite))
+        , marks_stream(storage.disk->writeFile(storage.marks_file_path, 4096, WriteMode::Append))
```

Comment:
It seems new marks should be appended to the older marks, not to replace them. And originally here was written `APPEND`, not `REWRITE`, but it seems https://github.com/ClickHouse/ClickHouse/pull/8862/files#diff-42d2ff7e53241a29f266116cea5661960ed5d771bc271066778df3c1c1a19e04R112 changed that.